controllers/customer-admin-controller/pkg/controllernamespace/namespace/namespace_controller.go

```diff
@@ -0,0 +1,137 @@
+package namespace
+
+import (
+	"context"
+	"log"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/openshift-azure/controllers/customer-admin-controller/pkg/common"
+)
+
+/**
+* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
+* business logic.  Delete these comments after modifying this file.*
+ */
+
+// Add creates a new CustomerAdminReconciler Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileNamespace{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("customeradminreconciler-namespace-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// TODO: figure out how to filter on namespaces that are !prefixed by a certain string
+	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileNamespace{}
+
+// ReconcileNamespace reconciles a CustomerAdminReconciler object
+type ReconcileNamespace struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile receives a namespace event and reconciles rolebindings for
+// customer roles
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNamespace) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if !common.RestrictedNamespace(request.Name) {
+		// move to appropriate place
+		// If we are namespace we need to reconcile 3 rolebindings:
+		// - customer-admin
+		// - customer-project-admin
+		// - customer-view
+		log.Printf("Reconciling Namespace %s/%s\n", request.Namespace, request.Name)
```

Comment:
You don't need \n with log. Also, should we start using our own log package or at least logrus? I'd go with logrus if we don't want to pull any deps from our repo. I am fine with switching to logrus in a follow-up.

Reply:
I didn't know about the `\n`, thanks.  I'm fine with logrus.

Reply:
updated to logrus.  also removed teh `\n`